MHCollapsibleViewManager.m

```diff
@@ -0,0 +1,228 @@
+//
+//  MHCollapsibleViewManager.m
+//  This is a manager of a uiview and view controller
+//
+//  Created by Lizzy Randall on 5/6/15.
+//  Copyright (c) 2015 Lizzy Randall. All rights reserved.
+//
+
+#import "MHCollapsibleViewManager.h"
+#import "MHFilterLabel.h"
+
+@implementation MHCollapsibleViewManager
+
+//@synthesize tableView;
+//@synthesize viewController;
+@synthesize rowAnimation;
+
+@synthesize data;
+@synthesize cellStyles;
+@synthesize headerTitle;
+
+@synthesize expanded;
+@synthesize numOfRows;
+
+//Initialize with animation, title and tableView
+- (id)initalizeManager:(UITableViewRowAnimation)animation title:(NSString*) title tableView:(UITableView*)tableView
+{
+    self.headerTitle = title;
+    self.rowAnimation = animation;
+    //initialized first and then can be overwritten by controller
+    //tableView is needed to create cells to store types
+    [self setCellProperties:nil header:nil clickedCell:nil normalCell:nil tableView:tableView];
+    return self;
+}
+
+//called after Initializing Manager
+- (void) initializeData:(NSMutableArray*)dataFromController
+{
+    self.data = [[NSMutableArray alloc] init];
+    for(int i=0; i < [dataFromController count] ; i++)
```

Comment:
As you might have noticed in the example above I used the array's enumeration method instead of a regular for loop. I would recommend this as it is easier to write (with the nice auto complete tool in the IDE), easier to read and is often more efficient.